### PR TITLE
Add an example that shows how to polyfill XHR in an extension

### PR DESCRIPTION
**Pull request by patrickkettner**
EOM

### CODE REVIEW

File: functional-samples/libraries-xhr-in-sw/manifest.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "Fetching Titles",
+  "version": "0.1",
+  "manifest_version": 3,
+  "description": "This extension fetches the titles of all the tabs in the current window and displays them in a side panel.",
+  "background": {
+    "service_worker": "dist/background.js"
+  },
+  "permissions": ["activeTab", "scripting", "sidePanel"],
```
Comment (sebastianbenz) [inline functional-samples/libraries-xhr-in-sw/manifest.json:9]: Do you need scripting permission?

File: functional-samples/libraries-xhr-in-sw/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Using XHR in Service Workers
+
+This sample demonstrates how to use code that relies on XHR within a extension's background service worker.
+
+## Overview
+
+The default background environment for extensions is the service worker. As a result, it only has direct access to [Fetch](https://developer.mozilla.org/docs/Web/API/Fetch_API/Using_Fetch). If you want to use a library that is built with XHR, this will not work by default. However, you can usually monkeypatch the expected behavior by polyfilling XHR. This sample shows an example of how you can use build tools to automatically inject a polyfill for XHR that covers most common XHR use cases, allowing for seamless integration into your extension.
```
Comment (sebastianbenz) [inline functional-samples/libraries-xhr-in-sw/README.md:7]: I think it'd be good to explain a bit more what you're doing here (including links to relevant sections).

File: functional-samples/libraries-xhr-in-sw/manifest.json (hunk 1 above)
Comment (patrickkettner) [reply functional-samples/libraries-xhr-in-sw/manifest.json:9]: nope! changed the demo and forgot to remove this. good catch!

File: functional-samples/libraries-xhr-in-sw/README.md (hunk 2 above)
Comment (patrickkettner) [reply functional-samples/libraries-xhr-in-sw/README.md:7]: added more

**Review by sebastianbenz — APPROVED** [review]
Thanks Patrick!

**Pull request merged**